openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -180,3 +206,60 @@ def _email_url_parser(url_name, extra_param=None):
     dashboard_url_path = reverse(url_name) + extra_param if extra_param else reverse(url_name)
     dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
     return urlparse.urlunparse(dashboard_link_parts)
+
+
+def get_credit_providers_by_course(user, course_key):
+    """Get the course information from ecommerce and parse the data to get providers.
+
+    Arguments:
+        user (User): The user object.
+        course_key (CourseKey): The identifier for the course.
+
+    Returns:
+        list, containing course providers.
+    """
+    credit_config = CreditConfig.current()
+
+    # Bypass caching for staff users, who may be adding providers and want
+    # to see them immediately.
+
+    providers_data = None
+    cache_key = None
+    if credit_config.enabled:
+        cache_key = '{key_prefix}.{course_key}'.format(
+            key_prefix=credit_config.CACHE_KEY, course_key=unicode(course_key)
+        )
+        providers_data = cache.get(cache_key)
+
+    if providers_data is not None:
+        return providers_data
+
+    providers_data = []
+    try:
+        response = ecommerce_api_client(user).courses(unicode(course_key) + '/?include_products=1').get()
```

Comment:
@awais786 in your example, the LMS is making requests to an endpoint accessible to regular users. In this case, requests are being made to an endpoint which is only accessible to staff users. This is the real problem I'm trying to point out.

[`get_edx_api_data`](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L14) is used to communicate with programs and credentials in situations like these. The only obstacle I can think of to using it here is that the ecommerce OAuth client's credentials aren't used to sign JWTs.